### ISSUE TITLE
CA1062 reports problems when accessing objects after using foreach

### ISSUE THREAD

**Issue by SWarnberg** [dotnet/roslyn-analyzers#2327]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.1

#### Diagnostic ID

[CA1062](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1062-validate-arguments-of-public-methods)

#### Repro steps

Consider this code:

```
public static class MainClass
{
  public static void Initialize(ModelsCollection models)
  {
    if (models == null) throw new ArgumentNullException(nameof(models));
    foreach (var item in models)
    {
      Debug.WriteLine(item.Name);
    }
    foreach (var item in models)
    {
      Debug.WriteLine(item.Id);
    }
  }
}

public class ModelsCollection : ObservableCollection<Model>
{
}

public class Model
{
  public int Id { get; set; }
  public string Name { get; set; }
}
```

#### Expected behavior
The code in **Initialize** should pass without warning.
#### Actual behavior
CA1062 is reported on the second foreach statement, telling that _models_ is not checked for null. This is a continuous problem, as soon as a foreach is used, every access to that object gets a CA1062.

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/2345
